### PR TITLE
[Book_Store] implemented JPA conception

### PR DESCRIPTION
**Pull request by fedorovychh**
extended JpaRepository interface
added Liquibase support
implemented soft delete concept for the Book entity
added missing endpoints and methods for BookService

### CODE REVIEW

File: src/main/java/com/app/bookstore/controller/BookController.java [hunk 1]
```diff
@@ -2,18 +2,23 @@
...
+import org.springframework.web.bind.annotation.PutMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
 
-@RequiredArgsConstructor
 @RestController
+@RequiredArgsConstructor
 @RequestMapping(value = "/api/books")
```
Comment (d1sam) [inline src/main/java/com/app/bookstore/controller/BookController.java]: you can set `/api` inside application.properties for all your controllers:
`server.servlet.context-path=/api`

File: src/main/java/com/app/bookstore/service/book/BookService.java [hunk 2]
```diff
@@ -10,4 +11,8 @@ public interface BookService {
     BookDto findById(Long id);
 
     List<BookDto> findAll();
+
+    void deleteBookById(Long id);
+
+    BookDto updateBookById(Long id, UpdateBookRequestDto updateBookRequestDto);
```
Comment (d1sam) [inline src/main/java/com/app/bookstore/service/book/BookService.java]:
```suggestion
    BookDto updateBookById(Long id, CreateBookRequestDto updateBookRequestDto);

```

Comment (d1sam) [reply src/main/java/com/app/bookstore/service/book/BookService.java]: you can use `CreateBookRequestDto` for update

File: src/main/java/com/app/bookstore/mapper/BookMapper.java [hunk 3]
```diff
@@ -11,4 +12,6 @@ public interface BookMapper {
     BookDto toDto(Book book);
 
     Book toBook(CreateBookRequestDto bookRequestDto);
+
+    Book toBook(UpdateBookRequestDto updateBookRequestDto);
```
Comment (d1sam) [inline src/main/java/com/app/bookstore/mapper/BookMapper.java]:
```suggestion
```

File: src/main/java/com/app/bookstore/service/book/BookServiceImpl.java [hunk 4]
```diff
@@ -38,4 +40,25 @@ public List<BookDto> findAll() {
...
+        book.setId(id);
+        return bookMapper.toDto(bookRepository.save(book));
+    }
+
+    private void checkIfBookPresent(Long id) {
+        Optional<Book> bookOptional = bookRepository.findById(id);
+        if (bookOptional.isEmpty()) {
+            throw new EntityNotFoundException("Can't find book by id: " + id);
+        }
+    }
```
Comment (d1sam) [inline src/main/java/com/app/bookstore/service/book/BookServiceImpl.java]:
```suggestion
```
you can use method `bookRepository.existsById()` as alternative

**Review by d1sam — CHANGES_REQUESTED** [review]
Well done, but some little changes should be done!

File: src/main/java/com/app/bookstore/repository/book/BookRepository.java [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package com.app.bookstore.repository.book;
+
+import com.app.bookstore.model.Book;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface BookRepository extends JpaRepository<Book, Long> {
```
Comment (bhdnchui) [inline src/main/java/com/app/bookstore/repository/book/BookRepository.java:8]: Add `@Repository`

File: src/main/java/com/app/bookstore/service/book/BookService.java [hunk 6]
```diff
@@ -10,4 +10,8 @@ public interface BookService {
     BookDto findById(Long id);
 
     List<BookDto> findAll();
+
+    void deleteBookById(Long id);
+
+    BookDto updateBookById(Long id, CreateBookRequestDto createBookRequestDto);
```
Comment (bhdnchui) [inline src/main/java/com/app/bookstore/service/book/BookService.java]: We can simplify variable name here.
```suggestion
    BookDto updateBookById(Long id, CreateBookRequestDto bookDto);
```

Comment (bhdnchui) [inline src/main/java/com/app/bookstore/service/book/BookService.java]: Also you will have only one update method, so we can just update.
```suggestion
    BookDto update(Long id, CreateBookRequestDto createBookRequestDto);
```

**Review by bhdnchui — CHANGES_REQUESTED** [review]
Just a few minor points to consider.

**Review by bhdnchui — APPROVED** [review]
Good job!

**Pull request merged**